server/src/main/java/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -846,16 +848,16 @@ public boolean configure(final String name, final Map<String, Object> xmlParams)
         _forceHA = Boolean.parseBoolean(value);
 
         value = params.get(TimeToSleep.key());
-        _timeToSleep = (long)NumbersUtil.parseInt(value, 60) * 1000;
+        _timeToSleep = NumbersUtil.parseLong(value, parseLong(TimeToSleep.defaultValue())) * 1000;
```

Comment:
since you have changed the type from Integer to Long, then second parseLong in these lines are not needed any more I think

Reply:
this is needed because return type is "string" and the method takes long.

Reply:
I am wondering; why did you use `defaultValue` instead of `value`? The default value is not updated when the configuration value changes. You can use `TimeToSleep.value()` that will return a Long.

My 2 cents would be to change it into something like:
```
private static final long MILISECONDS_TO_SECONDS_FACTOR = 1000

(...)

_timeToSleep = TimeToSleep.value() * MILISECONDS_TO_SECONDS_FACTOR
```
Same goes with `_maxRetries = MaxRetries.value()` etc.

Reply:
ok I thought, for the default value, I use the default method .. my fault..